packages/tokens/non-fungible/src/extensions/enumerable/storage.rs

```diff
@@ -0,0 +1,536 @@
+///   owner.
+/// * `from` - The account whose token is destroyed.
+/// * `token_id` - The token to burn.
+///
+/// # Errors
+///
+/// * refer to [`crate::burnable::burn`] errors.
+/// * refer to [`remove_from_owner_enumeration`] errors.
+///
+/// # Events
+///
+/// * topics - `["burn", from: Address]`
+/// * data - `[token_id: TokenId]`
+///
+/// # Notes
+///
+/// This is a wrapper around [`crate::burnable::burn()`], that also
+/// handles the storage updates for:
+/// * total supply
+/// * owner_tokens enumeration
+pub fn sequential_burn_from(e: &Env, spender: &Address, from: &Address, token_id: TokenId) {
+    burn_from(e, spender, from, token_id);
+
+    remove_from_owner_enumeration(e, from, token_id);
+
+    // We don't need the total supply, we just need to increment it.
+    let _ = decrement_total_supply(e);
+
+    // We don't need to update the global lists, `token_id`s act as the global
+    // index in sequential minting.
+}
+
+/// Destroys the `token_id` from `account`, by using `spender`s approval.
+///
+/// # Arguments
+///
+/// * `e` - Access to the Soroban environment.
+/// * `spender` - The account that is allowed to burn the token on behalf of the
+///   owner.
+/// * `from` - The account whose token is destroyed.
+/// * `token_id` - The token to burn.
+///
+/// # Errors
+///
+/// * refer to [`crate::burnable::burn`] errors.
+/// * refer to [`remove_from_owner_enumeration`] errors.
+/// * refer to [`remove_from_global_enumeration`] errors.
+///
+/// # Events
+///
+/// * topics - `["burn", from: Address]`
+/// * data - `[token_id: TokenId]`
+///
+/// # Notes
+///
+/// This is a wrapper around [`crate::burnable::burn()`], that also
+/// handles the storage updates for:
+/// * total supply
+/// * owner_tokens enumeration
+/// * global_tokens enumeration
+pub fn non_sequential_burn_from(e: &Env, spender: &Address, from: &Address, token_id: TokenId) {
+    burn_from(e, spender, from, token_id);
+
+    remove_from_owner_enumeration(e, from, token_id);
+
+    let total_supply = decrement_total_supply(e);
+
+    remove_from_global_enumeration(e, token_id, total_supply);
+}
+
+/// Transfers a non-fungible token (NFT), ensuring ownership checks.
+///
+/// # Arguments
+///
+/// * `e` - The environment reference.
+/// * `from` - The current owner's address.
+/// * `to` - The recipient's address.
+/// * `token_id` - The identifier of the token being transferred.
+///
+/// # Errors
+///
+/// * refer to [`crate::storage::transfer`] errors.
+/// * refer to [`remove_from_owner_enumeration`] errors.
+///
+/// # Events
+///
+/// * topics - `["transfer", from: Address, to: Address]`
+/// * data - `[token_id: TokenId]`
+///
+/// # Notes
+///
+/// This is a wrapper around [`crate::storage::transfer`], that also
+/// handles the storage updates for:
+/// * owner_tokens enumeration
+pub fn transfer(e: &Env, from: &Address, to: &Address, token_id: TokenId) {
+    base_transfer(e, from, to, token_id);
```

Comment:
No strong opinion, but if the goal is clarity, wouldn't it be better to use the fully-qualified function than an alias?